### PR TITLE
Add flex attention flags and args to Llama on Habana

### PR DESCRIPTION
**Pull request by AKloniecki**
# What does this PR do?

This is a cherry-pick of PR #2124 with tests added.

Flex attention is an attention function, alternative to flash attention.

Example usage:
Consider current command using flash attention:
`PT_HPU_LAZY_MODE=1  python3  run_generation.py    --model_name_or_path meta-llama/Llama-3.1-8B-Instruct  --attn_softmax_bf16 --use_hpu_graphs --limit_hpu_graphs --use_kv_cache --max_new_tokens 8192 --bf16 --batch_size 100  --use_flash_attention --flash_attention_recompute  --bucket_size=128 --bucket_internal    --trim_logits --max_input_tokens 128  --warmup 2`

it may be computed using flex attention as follows:
`PT_HPU_LAZY_MODE=1  python3  run_generation.py    --model_name_or_path meta-llama/Llama-3.1-8B-Instruct  --attn_softmax_bf16 --use_hpu_graphs --limit_hpu_graphs --use_kv_cache --max_new_tokens 8192 --bf16 --batch_size 100  --use_flex_attention --bucket_size=128 --bucket_internal    --trim_logits --max_input_tokens 128  --warmup 2`

### CODE REVIEW

**Comment by astachowiczhabana** [comment]
@AKloniecki please check

ruff check . setup.py
All checks passed!
ruff format --check . setup.py
Would reformat: tests/test_text_generation_example.py
1 file would be reformatted, 468 files already formatted
make: *** [Makefile:29: style_check] Error 1
test test_methods failed to style_check with exit status 2
failed executing optimum-habana-fork tests
~

**Review by astachowiczhabana — APPROVED** [review]

**Comment by astachowiczhabana** [comment]
Hi @regisss 
Can you please merge?
